frontend/src/pages/InsideFolder.js

```diff
@@ -1,32 +1,27 @@
-import React, { useState, Component, useEffect } from 'react';
-import Header from '../main-components/Header';
-import Sidebar from '../main-components/Sidebar';
-import InsideFolderContent from '../main-components/InsideFolderContent';
-import queryString from 'query-string';
+import React, { useState, Component, useEffect } from "react";
+import Header from "../main-components/Header";
+import Sidebar from "../main-components/Sidebar";
+import InsideFolderContent from "../main-components/InsideFolderContent";
+import queryString from "query-string";
 
 function InsideFolder(props) {
-	const [sideSetting, setSideSetting] = useState('f');
-	const handleClick = (e) => {
-		if (sideSetting === 'f') {
-			setSideSetting('t');
-		} else {
-			setSideSetting('f');
-		}
-	};
+  const [sidebarVisibility, setSidebarVisibility] = useState(false);
+  const handleClick = (e) => {
+    setSidebarVisibility((sidebarVisibility) => !sidebarVisibility);
```

Comment:
nit: This code is repeated across a few js scripts. Wonder if there isn't way to template this out?

Reply:
I tried to template it out to a seperate js file but this creates more problems then fixes. The reason being that each page has their own instance of sidebarVisibiltiy and setSidebarVisibility. If you make a function that takes these two as parameters it creates an infinite loop for some weird reason.

Reply:
```jsx
function ContentWithSidebar(props) {
  const { children } = props;
  const [sidebarVisibility, setSidebarVisibility] = useState(false);
  const handleClick = (e) => {
    setSidebarVisibility((sidebarVisibility) => !sidebarVisibility);
  };

  return (
    <div className='App'>
      <Header id='head' handleClick={handleClick}></Header>
      <div id='main'>
        <Sidebar visible={sidebarVisibility}></Sidebar>
        {children}
      </div>
    </div>
  );
}
```

Usage:

```
function InsideFolder(props) {
  const values = queryString.parse(props.location.search);
  const folderKey = values.folderKey;

  return (
    <ContentWithSidebar>
      <InsideFolderContent folderKey={folderKey}></InsideFolderContent>
    </ContentWithSidebar>
  );
};
```
  );
}
```

Reply:
```jsx
function ContentWithSidebar(props) {
  const { children } = props;
  const [sidebarVisibility, setSidebarVisibility] = useState(false);
  const handleClick = (e) => {
    setSidebarVisibility((sidebarVisibility) => !sidebarVisibility);
  };

  return (
    <div className='App'>
      <Header id='head' handleClick={handleClick}></Header>
      <div id='main'>
        <Sidebar visible={sidebarVisibility}></Sidebar>
        {children}
      </div>
    </div>
  );
}
```

Usage:

```js
function InsideFolder(props) {
  const values = queryString.parse(props.location.search);
  const folderKey = values.folderKey;

  return (
    <ContentWithSidebar>
      <InsideFolderContent folderKey={folderKey}></InsideFolderContent>
    </ContentWithSidebar>
  );
};
```

Reply:
I wasn't making another component but was just using a regular javascript function. This makes a lot more sense.